### PR TITLE
Chronicle - [Google Chronicle Backstory]

### PR DESCRIPTION
**Pull request by crestdatasystems**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
- Updated the client with the Google oauth2 library.
- Added support for the commands and fetch incidents mechanism to do up to 3 internal retries with a gap of 15, 30, and 60 seconds (exponentially) between the retries.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/GoogleChronicleBackstory/Integrations/GoogleChronicleBackstory/GoogleChronicleBackstory.py [hunk 1]
```diff
@@ -3,22 +3,23 @@
...
-# A request will be tried 3 times if it fails at the socket/connection level
-httplib2.RETRIES = 3
-
 ''' CONSTANTS '''
 
 DATE_FORMAT = '%Y-%m-%dT%H:%M:%S.%fZ'
 
 SCOPES = ['https://www.googleapis.com/auth/chronicle-backstory']
+STATUS_LIST_TO_RETRY = [429] + [i for i in range(500, 600)]
+MAX_RETRIES = 4
```
Comment (shmuel44) [inline Packs/GoogleChronicleBackstory/Integrations/GoogleChronicleBackstory/GoogleChronicleBackstory.py:21]: Why did you choose 4?

File: Packs/GoogleChronicleBackstory/Integrations/GoogleChronicleBackstory/GoogleChronicleBackstory.py [hunk 2]
```diff
@@ -197,37 +214,84 @@ def __init__(self, params: Dict[str, Any], proxy, disable_ssl):
...
+                https_adapter = http_adapter
+            elif IS_PY3 and PY_VER_MINOR >= 10:
+                https_adapter = SSLAdapter(max_retries=retry, verify=not self.disable_ssl)  # type: ignore[arg-type]
+            else:
+                https_adapter = http_adapter
+
+            self.http_client.mount('https://', https_adapter)
+
+        except NameError:
+            pass
```
Comment (shmuel44) [inline Packs/GoogleChronicleBackstory/Integrations/GoogleChronicleBackstory/GoogleChronicleBackstory.py:288]: I see that this method is copied from the `baseClient` class in the `commonServerPython` file. Could we just inherit it? (`class Client(BaseClient):`)

Please feel free to reach out to me with any questions 
I'm available here or on slack.

**Review by shmuel44 — COMMENTED** [review]
Hi @crestdatasystems ,
Thank you for your contribution!
Good work 🥇

File: Packs/GoogleChronicleBackstory/Integrations/GoogleChronicleBackstory/GoogleChronicleBackstory.py [hunk 3]
```diff
@@ -197,37 +214,84 @@ def __init__(self, params: Dict[str, Any], proxy, disable_ssl):
...
+                         backoff_factor=5,
+                         raise_on_redirect=False,
+                         raise_on_status=False):
+        """
+        Implements the retry mechanism.
+        In the default case where retries = 0 the request will fail on the first time.
 
-''' HELPER FUNCTIONS '''
+        :type retries: ``int``
+        :param retries: How many retries should be made in case of a failure. when set to '0'- will fail on the first time.
```
Comment (shmuel44) [inline Packs/GoogleChronicleBackstory/Integrations/GoogleChronicleBackstory/GoogleChronicleBackstory.py:228]: I see that this method is copied from the baseClient class in the commonServerPython file. Could we just inherit it? (class Client(BaseClient):)

Please feel free to reach out to me with any questions
I'm available here or on slack.

Comment (crestdatasystems) [reply Packs/GoogleChronicleBackstory/Integrations/GoogleChronicleBackstory/GoogleChronicleBackstory.py:228]: The reason for not using BaseClient was to create the authorized session through the Google Auth2 library, where BaseClient creates its own session which is not required here.

**Review by shmuel44 — APPROVED** [review]
Good Work!!

**Comment by shmuel44** [comment]
Hi @crestdatasystems,
I merged the internal PR, and your update is now in the master branch of content repo.
Your update should be available in the marketplace within the next 24 hours.
Again, thank you very much for your contribution!

**Pull request merged**